src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -682,6 +699,45 @@ CreateHashDistributedTable(Oid relationId, char *distributionColumnName,
 
 
 /*
+ * EnsureSchemaExistsOnAllNodes connects to all nodes with citus extension user
+ * and creates the schema of the given relationId. The function errors out if the
+ * command cannot be executed in any of the worker nodes.
+ */
+static void
+EnsureSchemaExistsOnAllNodes(Oid relationId)
+{
+	List *workerNodeList = ActiveWorkerNodeList();
+	ListCell *workerNodeCell = NULL;
+	StringInfo applySchemaCreationDDL = makeStringInfo();
+
+	Oid schemaId = get_rel_namespace(relationId);
+	char *schemaName = get_namespace_name(schemaId);
+	const char *createSchemaDDL = CreateSchemaDDLCommand(schemaId);
+
+	if (createSchemaDDL != NULL)
+	{
+		appendStringInfo(applySchemaCreationDDL, "%s", createSchemaDDL);
+
+		foreach(workerNodeCell, workerNodeList)
+		{
+			WorkerNode *workerNode = (WorkerNode *) lfirst(workerNodeCell);
+			char *nodeName = workerNode->workerName;
+			uint32 nodePort = workerNode->workerPort;
+			bool commandExecuted = false;
+
+			commandExecuted = ExecuteRemoteCommand(nodeName, nodePort,
```

Comment:
I realise `ExecuteRemoteCommand` has a convenient signature, but it calls legacy `MultiClient*` functions which we'd like to phase out. Could maybe use `SendCommandListToWorkerInSingleTransaction` here or, since it's a single command, `GetNodeUserDatabaseConnection(FORCE_NEW_CONNECTION, ...)` + `ExecuteCriticalRemoteCommand`, or a new wrapper for that.